cmd/sealer/cmd/cluster/rollback.go

```diff
@@ -0,0 +1,251 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+
+	"github.com/sealerio/sealer/cmd/sealer/cmd/types"
+	"github.com/sealerio/sealer/cmd/sealer/cmd/utils"
+	"github.com/sealerio/sealer/pkg/application"
+	clusterruntime "github.com/sealerio/sealer/pkg/cluster-runtime"
+	"github.com/sealerio/sealer/pkg/clusterfile"
+	imagev1 "github.com/sealerio/sealer/pkg/define/image/v1"
+	"github.com/sealerio/sealer/pkg/define/options"
+	"github.com/sealerio/sealer/pkg/imagedistributor"
+	"github.com/sealerio/sealer/pkg/imageengine"
+	"github.com/sealerio/sealer/pkg/infradriver"
+)
+
+var rollbackFlags *types.RollbackFlags
+
+var longRollbackCmdDescription = `rollback command is used to rollback a Kubernetes cluster via specified Clusterfile.`
+
+var exampleForRollbackCmd = `
+  sealer rollback docker.io/sealerio/kubernetes:v1.22.15-rollback
+`
+
+func NewRollbackCmd() *cobra.Command {
+	rollbackCmd := &cobra.Command{
+		Use:     "rollback",
+		Short:   "rollback a Kubernetes cluster via specified Clusterfile",
+		Long:    longRollbackCmdDescription,
+		Example: exampleForRollbackCmd,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				err         error
+				clusterFile = rollbackFlags.ClusterFile
+			)
+			if len(args) == 0 && clusterFile == "" {
+				return fmt.Errorf("you must input image name Or use Clusterfile")
+			}
+
+			if clusterFile != "" {
+				return rollbackWithClusterfile(clusterFile)
+			}
+
+			imageEngine, err := imageengine.NewImageEngine(options.EngineGlobalConfigurations{})
+			if err != nil {
+				return err
+			}
+
+			id, err := imageEngine.Pull(&options.PullOptions{
+				Quiet:      false,
+				PullPolicy: "missing",
```

Comment:
use constant here?

Reply:
same problem in run.go，recommend to solve in one branch together。